ndscan/plots/model/online_analysis.py

```diff
@@ -1,9 +1,12 @@
 import asyncio
+import logging
 from concurrent.futures import ProcessPoolExecutor
 from pyqtgraph import SignalProxy
 from qasync import QtCore
-from typing import Any, Dict
-from ...utils import FIT_OBJECTS
+from ...experiment import FitDescription
```

Comment:
As mentioned in the discussion thread, I'd be keen to keep the clean separation between `experiment` and other packages. `experiment` is allowed to import ARTIQ, including the whole dependencies on the coredevice stuff (pulling in the compiler dependencies like llvmlite, etc.), whereas the rest isn't (well, except for the `artiq.{dashboard, gui}` for the dashboard, and `artiq.applets.simple` for the applet.

I'm not opposed to documenting the structure of the metadata (schemata/…) more explicitly through dataclasses, but imho those should be in some sort of extra package. Currently, `ndscan.utils` is the dumping ground for shared things, but for the schema stuff, an extra package may be more appropriate.

Reply:
:+1: I wonder if it's worth having a schemata package then?